File: 0242-valid-anagram/solution.java
import java.util.*;

class Solution {
    public boolean isAnagram(String s, String t) {

        if(mapper(s).equals(mapper(t))){
            return true;
        }

        return false;
    }

    private Map<Character, Integer> mapper(String a){
        Map<Character, Integer> map = new HashMap<>();

        for(char ch : a.toCharArray()){
            map.put(ch, map.getOrDefault(ch,0)+1);
        }
        return map;
    }
}

File: 0242-valid-anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        sMap = {}
        for ch in s:
            sMap[ch] = sMap.get(ch,0) + 1
        tMap = {}
        for ch in t:
            tMap[ch] = tMap.get(ch,0) + 1
        if sMap == tMap:
            return True
        return False
        
